Simple pattern causes perl panic

```
$ perl -e 'if ("xx" =~ /[^\S\W]{6}/) { print "yes >$&<\n"; } else { print "no \n"; }'
panic: regrepeat() called with unrecognized node type 99='OPFAIL' at -e line 1.

**Description**
<!-- A clear and concise description of what the bug is. -->
I think the above output says it all.

**Steps to Reproduce**
<!-- A one-liner or script to reproduce the issue. -->
See above.

**Expected behavior**
<!-- A clear and concise description of what you expected to happen. -->
It should output "no".

**Perl configuration**
<!-- Please paste `perl -V` output just below. -->
```Summary of my perl5 (revision 5 version 38 subversion 2) configuration:
   
  Platform:
    osname=linux
    osvers=5.12.15-arch1-1
    archname=x86_64-linux-thread-multi
    uname='archlinux'
    config_args='-des -Dusethreads -Duseshrplib -Doptimize=-march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions         -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security         -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/perl/src=/usr/src/debug/perl -flto=auto -Dprefix=/usr -Dvendorprefix=/usr -Dprivlib=/usr/share/perl5/core_perl -Darchlib=/usr/lib/perl5/5.38/core_perl -Dsitelib=/usr/share/perl5/site_perl -Dsitearch=/usr/lib/perl5/5.38/site_perl -Dvendorlib=/usr/share/perl5/vendor_perl -Dvendorarch=/usr/lib/perl5/5.38/vendor_perl -Dscriptdir=/usr/bin/core_perl -Dsitescript=/usr/bin/site_perl -Dvendorscript=/usr/bin/vendor_perl -Dinc_version_list=none -Dman1ext=1perl -Dman3ext=3perl -Dlddlflags=-shared -Wl,-O1 -Wl,--sort-common -Wl,--as-needed -Wl,-z,relro -Wl,-z,now          -Wl,-z,pack-relative-relocs -flto=auto -Dldflags=-Wl,-O1 -Wl,--sort-common -Wl,--as-needed -Wl,-z,relro -Wl,-z,now          -Wl,-z,pack-relative-relocs -flto=auto -Dloclibpth=/usr/lib/db5.3 -Dlocincpth=/usr/include/db5.3'
    hint=recommended
    useposix=true
    d_sigaction=define
    useithreads=define
    usemultiplicity=define
    use64bitint=define
    use64bitall=define
    uselongdouble=undef
    usemymalloc=n
    default_inc_excludes_dot=define
  Compiler:
    cc='cc'
    ccflags ='-D_REENTRANT -D_GNU_SOURCE -fwrapv -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/include/db5.3 -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64'
    optimize='-march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/perl/src=/usr/src/debug/perl -flto=auto'

    cppflags='-D_REENTRANT -D_GNU_SOURCE -fwrapv -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/include/db5.3'
    ccversion=''
    gccversion='13.2.1 20230801'
    gccosandvers=''
    intsize=4
    longsize=8
    ptrsize=8
    doublesize=8
    byteorder=12345678
    doublekind=3
    d_longlong=define
    longlongsize=8
    d_longdbl=define
    longdblsize=16
    longdblkind=3
    ivtype='long'
    ivsize=8
    nvtype='double'
    nvsize=8
    Off_t='off_t'
    lseeksize=8
    alignbytes=8
    prototype=define
  Linker and Libraries:
    ld='cc'
    ldflags ='-Wl,-O1 -Wl,--sort-common -Wl,--as-needed -Wl,-z,relro -Wl,-z,now -Wl,-z,pack-relative-relocs -flto=auto -fstack-protector-strong -L/usr/lib/db5.3'
    libpth=/usr/local/lib /usr/lib /usr/lib/db5.3
    libs=-lpthread -lgdbm -ldb -ldl -lm -lcrypt -lutil -lc -lgdbm_compat
    perllibs=-lpthread -ldl -lm -lcrypt -lutil -lc
    libc=/lib/../lib/libc.so.6
    so=so
    useshrplib=true
    libperl=libperl.so
    gnulibc_version='2.39'
  Dynamic Linking:
    dlsrc=dl_dlopen.xs
    dlext=so
    d_dlsymun=undef
    ccdlflags='-Wl,-E -Wl,-rpath,/usr/lib/perl5/5.38/core_perl/CORE'
    cccdlflags='-fPIC'
    lddlflags='-shared -Wl,-O1 -Wl,--sort-common -Wl,--as-needed -Wl,-z,relro -Wl,-z,now -Wl,-z,pack-relative-relocs -flto=auto -L/usr/lib/db5.3 -fstack-protector-strong'
Characteristics of this binary (from libperl): 
  Compile-time options:
    HAS_LONG_DOUBLE
    HAS_STRTOLD
    HAS_TIMES
    MULTIPLICITY
    PERLIO_LAYERS
    PERL_COPY_ON_WRITE
    PERL_DONT_CREATE_GVSV
    PERL_HASH_FUNC_SIPHASH13
    PERL_HASH_USE_SBOX32
    PERL_MALLOC_WRAP
    PERL_OP_PARENT
    PERL_PRESERVE_IVUV
    PERL_USE_SAFE_PUTENV
    USE_64_BIT_ALL
    USE_64_BIT_INT
    USE_ITHREADS
    USE_LARGE_FILES
    USE_LOCALE
    USE_LOCALE_COLLATE
    USE_LOCALE_CTYPE
    USE_LOCALE_NUMERIC
    USE_LOCALE_TIME
    USE_PERLIO
    USE_PERL_ATOF
    USE_REENTRANT_API
    USE_THREAD_SAFE_LOCALE
  Built under linux
  Compiled at Feb 11 2024 19:15:41
  %ENV:
    PERLBREW_ROOT="/source/PerlBlead"
  @INC:
    /usr/lib/perl5/5.38/site_perl
    /usr/share/perl5/site_perl
    /usr/lib/perl5/5.38/vendor_perl
    /usr/share/perl5/vendor_perl
    /usr/lib/perl5/5.38/core_perl
    /usr/share/perl5/core_perl
```

FYI, it changed behavior somewhere between 5.32 and 5.34:

```
  --- perl5.30.3 --- 
no 

  --- perl5.32.1 --- 
no 

  --- perl5.34.3 --- 
panic: regrepeat() called with unrecognized node type 98='OPFAIL' at -e line 1.

  --- perl5.36.3 --- 
panic: regrepeat() called with unrecognized node type 98='OPFAIL' at -e line 1.

  --- perl5.38.2 --- 
panic: regrepeat() called with unrecognized node type 99='OPFAIL' at -e line 1.

  --- perl5.39.9 --- 
panic: regrepeat() called with unrecognized node type 99='OPFAIL' at -e line 1.

```

--- a/regcomp.c
+++ b/regcomp.c
@@ -12185,6 +12185,7 @@ S_optimize_regclass(pTHX_
 
   return_OPFAIL:
     op = OPFAIL;
+    *flagp &= ~(SIMPLE|HASWIDTH);
     *ret = reg1node(pRExC_state, op, 0);
     return op;
 

--- a/regcomp.c
+++ b/regcomp.c
@@ -17,7 +17,7 @@
  * debugging support added, which makes "use re 'debug'" work.
  */
 
-/* NOTE: this is derived from Henry Spencer's regexp code, and should not
+/* NOTE: this is derived from Henry Spencer's regexp code, and should not be
  * confused with the original package (see point 3 below).  Thanks, Henry!
  */
 
@@ -689,7 +689,7 @@ S_concat_pat(pTHX_ RExC_state_t * const pRExC_state,
                 oplist = OpSIBLING(oplist); /* skip CONST */
                 assert(oplist);
             }
-            oplist = OpSIBLING(oplist);;
+            oplist = OpSIBLING(oplist);
         }
 
         /* apply magic and QR overloading to arg */
@@ -11183,7 +11183,7 @@ S_regclass(pTHX_ RExC_state_t *pRExC_state, I32 *flagp, U32 depth,
 
         /* If optimized to something else and emitted, clean up and return */
         if (ret >= 0) {
-            SvREFCNT_dec(cp_list);;
+            SvREFCNT_dec(cp_list);
             SvREFCNT_dec(only_utf8_locale_list);
             SvREFCNT_dec(upper_latin1_only_utf8_matches);
             return ret;
@@ -11253,7 +11253,7 @@ S_regclass(pTHX_ RExC_state_t *pRExC_state, I32 *flagp, U32 depth,
                    : NULL,
                   only_utf8_locale_list);
 
-    SvREFCNT_dec(cp_list);;
+    SvREFCNT_dec(cp_list);
     SvREFCNT_dec(only_utf8_locale_list);
     return ret;
 }
@@ -11724,7 +11724,7 @@ S_optimize_regclass(pTHX_
              * convert to UTF-8 if not already there */
             if (value > 255) {
                 if (! UTF) {
-                    SvREFCNT_dec(cp_list);;
+                    SvREFCNT_dec(cp_list);
                     REQUIRE_UTF8(flagp);
                 }
 
@@ -11852,7 +11852,7 @@ S_optimize_regclass(pTHX_
             {
                 U8 ANYOFM_mask;
 
-                op = ANYOFM + inverted;;
+                op = ANYOFM + inverted;
 
                 /* We need to make the bits that differ be 0's */
                 ANYOFM_mask = ~ bits_differing; /* This goes into FLAGS */
@@ -15593,7 +15593,7 @@ S_parse_uniprop_string(pTHX_
                  * property hasn't been encountered yet, but at runtime, it's
                  * an error to try to use an undefined one */
                 if (! deferrable) {
-                    goto unknown_user_defined;;
+                    goto unknown_user_defined;
                 }
 
                 goto definition_deferred;

--- a/t/re/re_tests
+++ b/t/re/re_tests
@@ -2157,6 +2157,13 @@ AB\s+\x{100}	AB \x{100}X	y	-	-
 /^(xa|(?:[Z=])*\1a){2}$/	xa=xaaa	n	-	-	# GH 10073 - RT72020
 /^(xa|(?:[Z=]|zzzz)*\1a){2}$/	xa=xaaa	n	-	-	# GH 10073 - RT72020
 
+# GH 22094 - a character class optimized to OPFAIL shouldn't panic when quantified
+[^\W\S]{6}	a	n	-	-
+[^\W\S]{1,3}	a	n	-	-
+[^\W\S]+	a	n	-	-
+[^\W\S]*	a	y	$&	
+[^\W\S]?	a	y	$&	
+
 # Keep these lines at the end of the file
 # pat	string	y/n/etc	expr	expected-expr	skip-reason	comment
 # vim: softtabstop=0 noexpandtab

